modules/ding_react/plugins/content_types/related_materials.inc

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/**
+ * @file
+ * Add the checklist app content type.
+ */
+
+$plugin = [
+  'title' => t('Related materials app'),
+  'single' => TRUE,
+  'description' => t('Display related materials for an object.'),
+  'category' => t('DDB React'),
+  'required context' => [
+    new ctools_context_optional(t('Ting object'), 'ting_object'),
+    new ctools_context_optional(t('Ting collection'), 'ting_collection'),
+  ],
+];
+
+/**
+ * Return pane content.
+ */
+function ding_react_related_materials_content_type_render($subtype, $conf, $panel_args, $contexts) {
+  $block = new stdClass();
+
+  // We support multiple contexts which can result in a Ting object.
+  // Find the ones available and extract an object accordingly.
+  $contexts_with_data = array_filter($contexts,
+    function (ctools_context $context) {
+      return !empty($context->data);
+    }
+  );
+  $ting_objects = array_map(function (ctools_context $context) {
+    $data = $context->data;
+    if ($data instanceof TingCollection) {
+      return $data->getPrimary_object()->getTingObject();
+    }
+    elseif ($data instanceof TingEntity) {
+      return $data->getTingObject();
+    }
+  }, $contexts_with_data);
+  $object = array_shift($ting_objects);
+
+  if ($object) {
+    // Support setting allowed sources through a variable in case our hard-coded
+    // list is not sufficient. NB: There is currently no admin UI for setting
+    // this.
+    $sources = variable_get('ding_react_related_materials_sources', implode(',', [
+      'bibliotekskatalog',
+      'ereolen',
+      'ereolen global',
+      'comics plus',
+      'ebook central',
+      'rbdigital magazines',
+    ]));
+    $sources = explode(',', $sources);
+
+    $data = [
+      'subjects' => ding_react_related_materials_format_strings($object->getSubjects()),
+      'categories' => ding_react_related_materials_format_strings($object->getAudience()),
+      'sources' => ding_react_related_materials_format_strings($sources),
+      'exclude-title' => ding_react_related_materials_format_string($object->getTitle()),
+      // We cannot use url() here as it will encode the colon in the placeholder.
+      'search-url' => '/search/ting/:query?sort=:sort',
+      'material-url' => '/ting/object/:pid',
+      'cover-service-url' => 'https://cover.dandigbib.org/api',
```

Comment:
Why is the cover end-point hardcoded into this and not fetched from configuration in the CMS backend.

Reply:
Configuration for this has been added in #1640. The two PRs are somewhat intertwined at the moment 😕 .